### PR TITLE
Add Python 3.12 support

### PR DESCRIPTION
**Pull request by FasterSpeeding**
one day:tm:

(This will fail, just opening this as a place to track the compatibility issues)

### CODE REVIEW

**Comment by davfsa** [comment]
Readme :)

**Comment by davfsa** [comment]
Re-trigger CI

File: hikari/internal/fast_protocol.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
     from typing_extensions import Self
 
 _Protocol: FastProtocolChecking = NotImplemented
-_IGNORED_ATTRS = typing.EXCLUDED_ATTRIBUTES + ["__qualname__", "__slots__"]
+_IGNORED_ATTRS = frozenset(tuple(typing.EXCLUDED_ATTRIBUTES) + ("__qualname__", "__slots__"))
```
Comment (yakMM) [inline hikari/internal/fast_protocol.py:35]:
```suggestion
_IGNORED_ATTRS = frozenset(typing.EXCLUDED_ATTRIBUTES) | {"__qualname__", "__slots__"}
```

Comment (davfsa) [reply hikari/internal/fast_protocol.py:35]: That could work too, the main reason for the tuple was a little simplificacion, as I thought that `|` a frozenset would yield a normal set, but it's apparently a frozen one too

```py
>>> a = frozenset({1,2,3}) | {4,5,6}
>>> a
frozenset({1, 2, 3, 4, 5, 6})
```

TIL!

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
-aiohttp~=3.8
+# Temporary!
+aiohttp==3.9.0rc0
```
Comment (davfsa) [inline requirements.txt:2]:
```suggestion
aiohttp~=3.9
```

**Review by davfsa — APPROVED** [review]
And we have liftoff!

**Pull request merged**